python-lib/lal/handlers/lal_handler.py

```diff
@@ -109,6 +139,11 @@ def get_batch(self):
 
         remaining = self.get_remaining()
         ids_batch = remaining[-BATCH_SIZE:]
+        with LALHandler.lock:
+            for i in ids_batch:
+                if i not in self.sample_by_user_reservation:
+                    reserved_until = datetime.now() + timedelta(minutes=int(BATCH_SIZE * BLOCK_SAMPLE_BY_USER_FOR_MINUTES))
```

Comment:
Maybe compute the `reserved_until` outside of the loop to have the same for the whole batch.

Reply:
aggree